### PR TITLE
fix: Abide to the DataOfferCreationRequest.PolicyEnum rule

### PR DESCRIPTION
**Pull request by ununhexium**
https://github.com/sovity/PMO-Software/issues/1587

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [ ] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [ ] I have updated the Community Edition [Postman-Collection](https://github.com/sovity/edc-ce/blob/main/docs/api/postman_collection.json) if I changed existing APIs or added new APIs (e.g. for Management-API or API-Wrapper)
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java [hunk 1]
```diff
@@ -66,45 +67,90 @@ private boolean isIdAvailable(DSLContext dsl, Table<?> table, TableField<?, Stri
...
+        checkPolicyIdAvailable(dsl, policyId);
+        checkContractDefinitionIdAvailable(dsl, contractDefinitionId);
 
         assetApiService.createAsset(dataOfferCreationRequest.getUiAssetCreateRequest());
 
         val maybeNewPolicy = Optional.ofNullable(dataOfferCreationRequest.getUiPolicyExpression());
 
         maybeNewPolicy.ifPresent(
-            policy -> policyDefinitionApiService.createPolicyDefinitionV2(new PolicyDefinitionCreateDto(commonId, policy)));
+            policy -> policyDefinitionApiService.createPolicyDefinitionV2(new PolicyDefinitionCreateDto(policyId, policy)));
```
Comment (kamilczaja) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java:107]: You create a policy if the policy expression is not `null`. So if the expression is `null`, you don't create a policy but still pass the ID to the contract definition. What happens if the policy with this ID does not exist?

**Review by kamilczaja — CHANGES_REQUESTED** [review]

**Comment by kamilczaja** [comment]
I may be completely wrong but I don't think this works out

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java [hunk 2]
```diff
@@ -66,45 +67,90 @@ private boolean isIdAvailable(DSLContext dsl, Table<?> table, TableField<?, Stri
...
-        if (!contractDefinitionIdExists) {
-            throw new InvalidRequestException("Contract definition with id %s already exists".formatted(commonId));
-        }
+        checkAssetIdAvailable(dsl, assetId);
+        checkContractDefinitionIdAvailable(dsl, contractDefinitionId);
+
+        assetApiService.createAsset(dataOfferCreationRequest.getUiAssetCreateRequest());
+
+        return createContractDefinition(assetId, policyId, contractDefinitionId);
+    }
```
[the remark this answers is not in the document]
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java:91]: Added a check that created the `always-true` policy if it doesn't exist.

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java (hunk 1 above)
Comment (ununhexium) [reply extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/data_offer/DataOfferPageApiService.java]: Added exception when the policy definition is missing.

**Review by kamilczaja — APPROVED** [review]
LGTM

**Pull request merged**